File: gamelogic/src/main/java/com/alta_v2/game/gamelogic/stage/event/StageEventHandler.java
package com.alta_v2.game.gamelogic.stage.event;

/**
 * Describes an event handler of stage.
 */
@FunctionalInterface
public interface StageEventHandler<T> {

    /**
     * Handles the event.
     *
     * @param data - the data related to event.
     */
    void handle(T data);

}

File: gamelogic/src/main/kotlin/com/alta_v2/game/gamelogic/stage/event/StageEventHandler.kt
package com.alta_v2.game.gamelogic.stage.event

/**
 * Describes an event handler of stage.
 */
interface StageEventHandler<T> {

    /**
     * Handles the event.
     *
     * @param data - the data related to event.
     */
    fun handle(data: T)

}

interface ChangeScreenHandler : StageEventHandler<ChangeStageEvent>
interface ChangeMapEventHandler : StageEventHandler<Void>